components/registry/org.wso2.carbon.registry.indexing/src/main/java/org/wso2/carbon/registry/indexing/indexer/IndexDocumentCreator.java

```diff
@@ -200,16 +200,25 @@ private void addTags() throws RegistryException {
         Tag[] tags;
         tags = registry.getTags(resourcePath);
         List<String> tagList = new ArrayList<>();
+        List<String> taxonomyList = new ArrayList<>();
         if (tags != null && tags.length > 0) {
             for (Tag tag : tags) {
-                tagList.add(tag.getTagName());
+                if (tag.getTagName().contains("/")) {
+                    taxonomyList.add(tag.getTagName());
+                } else {
+                    tagList.add(tag.getTagName());
+                }
+
```

Comment:
Shall we add a comment here? 
With a sample tag value and what are these 2 lists.